### PR TITLE
do not save any metadata from the headers in tar extract

### LINKED ISSUE #17434: Wrong content type with x-amz-meta-snowball-auto-extract
I have reasons to use the undocumented x-amz-meta-snowball-auto-extract functionality, which gives the ability of uploading many files in one request.

As I noted there are issues with the content type. If I upload a .js file as part of a tar-stream, I will get the content type "binary/octet-stream".
Otherwise it's "application/x-javascript".

I cannot find a solution to set the right content type. In my opinion it should be possible to configure a manual mapping.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
do not save any metadata from the headers in tar extract

## Motivation and Context
only preserve the same storage-class as incoming
request other than that rest of them must be
deduced.

## How to test this PR?
fixes #17434 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/object-handlers.go [hunk 1]
```diff
@@ -2111,36 +2111,12 @@ func (api objectAPIHandlers) PutObjectExtractHandler(w http.ResponseWriter, r *h
...
-
 	putObjectTar := func(reader io.Reader, info os.FileInfo, object string) error {
 		size := info.Size()
 
-		// Copy metadata and make space for a bit more
-		metaCopy := make(map[string]string, len(metadata)+5)
-		for k, v := range metadata {
-			metaCopy[k] = v
+		metadata := map[string]string{
+			xhttp.AmzStorageClass: sc, // save same storage-class as incoming stream.
```
Comment (vadmeste) [inline cmd/object-handlers.go:2122]: 'sc' in this function can be empty, but usually there is no storage-class set in xl.meta when the user does not specify anything. 'mc stat' will show empty storage class if you set an empty sc here

File: cmd/object-handlers.go [hunk 2]
```diff
@@ -2111,36 +2111,12 @@ func (api objectAPIHandlers) PutObjectExtractHandler(w http.ResponseWriter, r *h
...
-	delete(metadata, xhttp.AmzSnowballExtract)
-	delete(metadata, xhttp.AmzSnowballExtract)
-	delete(metadata, xhttp.MinIOSnowballIgnoreDirs)
-	delete(metadata, xhttp.MinIOSnowballIgnoreErrors)
-	delete(metadata, xhttp.MinIOSnowballPrefix)
-
-	// Use the same storage class as the original uploaded tar file
-	if sc != "" {
-		metadata[xhttp.AmzStorageClass] = sc
-	}
```
Comment (poornas) [inline cmd/object-handlers.go:2132]: shouldn't we hold on to user defined metadata?

Comment (harshavardhana) [reply cmd/object-handlers.go:2132]: no for tar extract why? - do all objects have same metadata?

**Review by poornas — APPROVED** [review]

**Review by donatello — APPROVED** [review]

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**